onnxscript/rewriter/_ir_utils.py

```diff
@@ -10,24 +12,41 @@
 
 
 def propagate_const_value(ir_value: ir.Value) -> ir.Value:
+    """Temporary method to propagate a constant value to the IR value."""
     node = ir_value.producer()
     if ir_value.const_value is None and node is not None and node.op_type == "Constant":
-        attr_names = [
+        for attr_name in (
             "value_float",
             "value_int",
             "value_string",
             "value",
             "value_floats",
             "value_ints",
             "value_strings",
-        ]
-        for attr_name in attr_names:
+        ):
             attr_value = node.attributes.get(attr_name)
-            if attr_value is not None:
-                # TODO: RefAttr should be also supported?
-                if isinstance(attr_value, ir.Attr):
-                    ir_value.const_value = attr_value.value  # type: ignore[union-attr]
-                    break
+            if attr_value is None or not isinstance(attr_value, ir.Attr):
+                continue
+
+            if attr_name in {"value_float", "value_floats"}:
+                const_value = ir.Tensor(
+                    np.array(attr_value.value, dtype=np.float32), name=ir_value.name
+                )
+            elif attr_name in {"value_int", "value_ints"}:
+                const_value = ir.Tensor(
+                    np.array(attr_value.value, dtype=np.int64), name=ir_value.name
+                )
+            elif attr_name in {"value_string", "value_strings"}:
+                const_value = ir.StringTensor(
+                    np.array(attr_value.value, dtype=np.bytes_), name=ir_value.name
+                )
+            elif attr_name == "value":
+                const_value = typing.cast(ir.TensorProtocol, attr_value.value)
+
+            ir_value.const_value = const_value
```

Comment:
Should we reassign shape and dtype when value is assigned? Then we can avoid the situation that users forget to reassign shape/dtype when they update the const_value?

Reply:
Good idea

Reply:
But what happens when the original value has a symbolic shape? Do we want to narrow the symbols as concrete values when assigning constants?

Reply:
Reverted the change because initializers may not have the value information when deserializing, but this change will add value information to the initializer ir.Value objects, breaking the serde roundtrip. We can think about a better way of doing this.